src/flow/src/expr/relation/func.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::type_name;
+
+use common_time::{Date, DateTime};
+use datatypes::prelude::ConcreteDataType;
+use datatypes::value::{OrderedF32, OrderedF64, Value};
+use serde::{Deserialize, Serialize};
+
+use crate::expr::error::{EvalError, TryFromValueSnafu, TypeMismatchSnafu};
+use crate::expr::relation::accum::Accum;
+use crate::repr::Diff;
+
+/// Aggregate functions that can be applied to a group of rows.
+///
+/// `Mean` function is deliberately not included as it can be computed from `Sum` and `Count`, whose state can be better managed.
+///
+/// type of the input and output of the aggregate function:
+///
+/// `sum(i*)->i64, sum(u*)->u64`
+///
+/// `count()->i64`
+///
+/// `min/max(T)->T`
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Serialize, Deserialize, Hash)]
+pub enum AggregateFunc {
+    MaxInt16,
+    MaxInt32,
+    MaxInt64,
+    MaxUInt16,
+    MaxUInt32,
+    MaxUInt64,
+    MaxFloat32,
+    MaxFloat64,
+    MaxBool,
+    MaxString,
+    MaxDate,
+    MaxDateTime,
```

Comment:
Do we need a `MaxTimestamp`?

Reply:
> Do we need a `MaxTimestamp`?

Also add min/max for `Time` `Duration` and `Interval`